src/main/java/org/gitlab/api/models/GitlabProject.java

```diff
@@ -449,4 +452,27 @@ public String getImportUrl() {
     public void setImportUrl(String importUrl) {
         this.importUrl = importUrl;
     }
+
+    public GitlabProject getForkedFrom() {
+        return forkedFrom;
+    }
+
+    public void setForkedFrom(GitlabProject forkedFrom) {
+        this.forkedFrom = forkedFrom;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        GitlabProject that = (GitlabProject) o;
+
+        return id != null ? id.equals(that.id) : that.id == null;
```

Comment:
Is it really right to say that two projects with null id are equal?  Maybe they're two different projects that we have yet to issue create commands for?

Reply:
you are right

Reply:
@novalis 
If id is null so the 2 projects are not equals ? or It need to compare project name + project namespace ?
Thank you for your advice.

Reply:
I like the idea of comparing project name and namespace.  I think I might do that in all cases -- not just if the ids are null.